api/nodes/permissions.py

```diff
@@ -63,6 +65,36 @@ def has_object_permission(self, request, view, obj):
             has_auth = parent_node.can_edit(auth)
             return has_auth
 
+
+class ContributorOrPublicForComments(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, (Node, Comment, CommentReport)), 'obj must be a Node, Comment or CommentReport, got {}'.format(obj)
+        auth = get_user_auth(request)
+        if isinstance(obj, CommentReport):
+            comment_id = request.parser_context['kwargs']['comment_id']
+            comment = Comment.load(comment_id)
+            if not comment:
+                raise NotFound
+            node = comment.node
+        elif isinstance(obj, Comment):
+            comment = obj
+            node = obj.node
+        elif isinstance(obj, Node):
+            node = obj
+
+        view_name = view.get_view_name()
+        is_report_view = (view_name == 'Comment Reports List' or view_name == 'Comment Report Detail')
```

Comment:
Makes sense to me. http://www.django-rest-framework.org/api-guide/permissions/#setting-the-permission-policy Explicit is better than magic.